src/iris_grib/_load_convert.py

```diff
@@ -1724,76 +1728,114 @@ def vertical_coords(section, metadata):
-                            scaleFactorOfSecondFixedSurface,
-                        )
-                        point = 0.5 * (first + second)
-                        bounds = [first, second]
-                else:
-                    point = unscale(
-                        scaledValueOfFirstFixedSurface, scaleFactorOfFirstFixedSurface
-                    )
-                    bounds = None
-                coord = DimCoord(
-                    point,
-                    standard_name=fixed_surface.standard_name,
-                    long_name=fixed_surface.long_name,
-                    units=fixed_surface.units,
-                    bounds=bounds,
-                )
-                if fixed_surface == fixed_surface_missing:
-                    coord.attributes["GRIB_fixed_surface_type"] = (
-                        typeOfFirstFixedSurface
-                    )
+                second = _get_surface_value(section, "Second")
+            if second is not None:
+                point = 0.5 * (first + second)
+                bounds = [first, second]
+        else:
+            bounds = None
+        if point is None:
+            return
+        if bounds and (
+            fixed_surface_first.standard_name != fixed_surface_second.standard_name
+            or fixed_surface_first.long_name != fixed_surface_second.long_name
+            or fixed_surface_first.units != fixed_surface_second.units
+        ):
+            # Create two coords, one for each bound. we aren't allowed None as a bound
+            # so rely solely on the point value
+            # first_bounds = [bounds[0], None]
+            coord = DimCoord(
+                first,
+                standard_name=fixed_surface_first.standard_name,
+                long_name=fixed_surface_first.long_name,
+                units=fixed_surface_first.units,
+                # bounds=first_bounds,
+            )
+            # Add the vertical coordinate to metadata aux coords.
+            metadata["aux_coords_and_dims"].append((coord, None))
+
+            # second_bounds = [None, bounds[1]]
+            coord = DimCoord(
+                second,
+                standard_name=fixed_surface_second.standard_name,
+                long_name=fixed_surface_second.long_name,
+                units=fixed_surface_second.units,
+                # bounds=second_bounds,
+            )
+            # Add the vertical coordinate to metadata aux coords.
+            metadata["aux_coords_and_dims"].append((coord, None))
+        else:
+            coord = DimCoord(
+                point,
+                standard_name=fixed_surface_first.standard_name,
+                long_name=fixed_surface_first.long_name,
+                units=fixed_surface_first.units,
+                bounds=bounds,
+            )
+            # Add the vertical coordinate to metadata aux coords.
+            metadata["aux_coords_and_dims"].append((coord, None))
```

Comment:
I think this would be easier for future developers to follow if it was abstracted into a convenience function. It looks like that could support all 3 coord creation calls pretty easily?

Reply:
Done